chain/manager.go

```diff
@@ -1001,44 +1007,56 @@ func (m *Manager) V2TransactionSet(basis types.ChainIndex, txn types.V2Transacti
 	return m.tipState.Index, append(parents, txn), nil
 }
 
-func (m *Manager) checkDupTxnSet(txns []types.Transaction, v2txns []types.V2Transaction) (types.Hash256, bool) {
+func (m *Manager) checkTxnSet(txns []types.Transaction, v2txns []types.V2Transaction) (bool, error) {
 	allInPool := true
-	checkPool := func(txid types.TransactionID) {
+	checkPool := func(txid types.TransactionID) types.TransactionID {
 		if allInPool {
 			if _, ok := m.txpool.indices[txid]; !ok {
 				allInPool = false
 			}
 		}
+		return txid
```

Comment:
Kinda surprised you didn't go all the way to save those extra 3 lines of code :P 
```go
_, ok := m.txpool.indices[txid]
allInPool = allInPool && ok
```

Reply:
I would, but Go isn't smart enough to skip the map lookup if `allInPool` is already `false`!

Same thing applies to `for` loops:
```go
var all bool
for _, x := range xs {
    all = all || x
}
```
In theory the compiler could break out of the loop as soon as `all = true`, but it doesn't; you have to write it yourself. Sad.